options/base_options.py

```diff
@@ -20,10 +20,10 @@ def __init__(self):
     def initialize(self, parser):
         """Define the common options that are used in both training and test."""
         # basic parameters
-        parser.add_argument('--dataroot', required=True, help='path to images (should have subfolders trainA, trainB, valA, valB, etc)')
+        parser.add_argument('--dataroot', required=False, help='path to images (should have subfolders trainA, trainB, valA, valB, etc)')
         parser.add_argument('--name', type=str, default='experiment_name', help='name of the experiment. It decides where to store samples and models')
-        parser.add_argument('--gpu_ids', type=str, default='0', help='gpu ids: e.g. 0  0,1,2, 0,2. use -1 for CPU')
-        parser.add_argument('--checkpoints_dir', type=str, default='./checkpoints', help='models are saved here')
+        parser.add_argument('--gpu_ids', type=str, default='-1', help='gpu ids: e.g. 0  0,1,2, 0,2. use -1 for CPU')
+        parser.add_argument('--checkpoints_dir', type=str, default='/tmp/model/checkpoints', help='models are saved here')
```

Comment:
These were changed because we can't import `Options` objects otherwise. For instance, having an option with `required=True` will fail because we don't have command-line arguments to parse (since this isn't running from CLI).

Same thing goes for the other arguments. @TechTeam12 if you think of a better solution that could still be used for both training and inference, please take a crack at it